flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnTestBase.java

```diff
@@ -599,14 +599,14 @@ public boolean accept(File dir, String name) {
     }
 
     public static boolean verifyStringsInNamedLogFiles(
-            final String[] mustHave, final String fileName) {
-        List<String> mustHaveList = Arrays.asList(mustHave);
-        File cwd = new File("target/" + YARN_CONFIGURATION.get(TEST_CLUSTER_NAME_KEY));
+            final String[] mustHave, final String applicationId, final String fileName) {
```

Comment:
I would suggest to pass in an `ApplicationID`. This gives a bit better type safety.